### PR TITLE
Add support for status options

### PR DESCRIPTION
**Pull request by spekke**
This will fix https://github.com/SwiftGit2/SwiftGit2/issues/133

### CODE REVIEW

File: SwiftGit2/Objects.swift [hunk 1]
```diff
@@ -222,3 +222,28 @@ public struct Tag: ObjectType, Hashable {
...
+	public static let recurseIgnoredDirs = StatusOptions(rawValue: GIT_STATUS_OPT_RECURSE_IGNORED_DIRS.rawValue)
+	public static let renamesHeadToIndex = StatusOptions(rawValue: GIT_STATUS_OPT_RENAMES_HEAD_TO_INDEX.rawValue)
+	public static let renamesIndexToWorkDir = StatusOptions(rawValue: GIT_STATUS_OPT_RENAMES_INDEX_TO_WORKDIR.rawValue)
+	public static let sortCasesSensitively = StatusOptions(rawValue: GIT_STATUS_OPT_SORT_CASE_SENSITIVELY.rawValue)
+	public static let sortCasesInSensitively = StatusOptions(rawValue: GIT_STATUS_OPT_SORT_CASE_INSENSITIVELY.rawValue)
+	public static let renamesFromRewrites = StatusOptions(rawValue: GIT_STATUS_OPT_RENAMES_FROM_REWRITES.rawValue)
+	public static let noRefresh = StatusOptions(rawValue: GIT_STATUS_OPT_NO_REFRESH.rawValue)
+	public static let updateIndex = StatusOptions(rawValue: GIT_STATUS_OPT_UPDATE_INDEX.rawValue)
+	public static let includeUnreadable = StatusOptions(rawValue: GIT_STATUS_OPT_INCLUDE_UNREADABLE.rawValue)
+	public static let includeUnreadableAsUntracked = StatusOptions(rawValue: GIT_STATUS_OPT_INCLUDE_UNREADABLE_AS_UNTRACKED.rawValue)
```
[the remark this answers is not in the document]
Comment (BilalReffas) [reply SwiftGit2/Objects.swift:248]: Can you resolve this please @spekke

[the remark this answers is not in the document]
Comment (UberJason) [reply SwiftGit2/Objects.swift:248]: Hey! Any progress on this? Seems like a quick fix that would allow for a really important enhancement. 🙂

**Comment by UberJason** [comment]
Hi! Any progress on this? Seems like it would fix a key issue and be a great enhancement!

File: SwiftGit2/Objects.swift [hunk 2]
```diff
@@ -222,3 +222,60 @@ public struct Tag: ObjectType, Hashable {
 		message = String(validatingUTF8: git_tag_message(pointer))!
 	}
 }
+
+public struct StatusOptions: OptionSet {
```
Comment (atom2ueki) [inline SwiftGit2/Objects.swift:226]: @spekke it's good to create a new file for this `StatusOptions` just like `CheckoutStrategy`

Comment (BilalReffas) [reply SwiftGit2/Objects.swift:226]: I agree 👍

**Review by BilalReffas — COMMENTED** [review]
Looks quite good, thanks 👍

As @atom2ueki mentioned could you please create a new file for `StatusOptions` ?

**Comment by spekke** [comment]
@BilalReffas @atom2ueki Fixed comment regarding extracting StatusOptions into a separate file.

**Comment by twa16** [comment]
@spekke Thanks for the PR. Merging! @UberJason FYI

**Pull request merged**